### DISCUSSION TITLE
Close client Connection with custom message

### DISCUSSION THREAD

**Discussion by ironydips** [uNetworking/uWebSockets.js#899] [Q&A] [answered]
hi, we have to close client connection from backend with specific code & message. Close Reasons can be varied

- token is invalid
- token expired
- too many requests

We are using `ws.end` to send custom code and message `ws.end(1001, 'token invalid')` but we are not receiving on client side on both `.on(error) and .on(close)`.  Can you pls guide us what are we missing here.

**Comment by uNetworkingAB** [comment]
You have to check with your client how to do it, we support sending error codes with websocket.end(integer, string)

**Reply by ironydips** [reply, marked as answer]
UPDATE: Ignore this. Just realised, we have to send custom Error Codes b/w 4000 to 4999. It's working fine now.

sorry for confusion, by client side I meant my front end code. We are writing both backend and front end. Frontend use plain websocket connection ` new WebSocket(url)`

whenever we send `ws.end(101, 'already too many connections')` message in `open` method at backend, `ws.on('error')` gets called at frontend with this error message:

<img width="605" alt="Screenshot 2023-05-07 at 1 29 53 AM" src="https://user-images.githubusercontent.com/2323273/236644318-f6d3d925-88e1-4f5e-a5ff-911fb480e1bd.png">
